DNSecure/ContentView.swift

```diff
@@ -13,44 +13,41 @@ struct ContentView {
     @Binding var servers: Resolvers
     @Binding var usedID: String?
     @State var isEnabled = false
-    @State var selection: Int?
+    @State var selectedServer: Resolver?
     @State var alertIsPresented = false
     @State var alertTitle = ""
     @State var alertMessage = ""
     @State var guideIsPresented = false
 
     func addNewDoTServer() {
-        self.servers.append(
-            .init(
-                name: "New",
-                configuration: .dnsOverTLS(DoTConfiguration())
-            )
+        let server = Resolver(
+            name: "New",
+            configuration: .dnsOverTLS(DoTConfiguration())
         )
-        self.selection = self.servers.count - 1
+        self.servers.append(server)
+        self.selectedServer = server
     }
 
     func addNewDoHServer() {
-        self.servers.append(
-            .init(
-                name: "New",
-                configuration: .dnsOverHTTPS(DoHConfiguration())
-            )
+        let server = Resolver(
+            name: "New",
+            configuration: .dnsOverHTTPS(DoHConfiguration())
         )
-        self.selection = self.servers.count - 1
+        self.servers.append(server)
+        self.selectedServer = server
     }
 
     func removeServers(at indexSet: IndexSet) {
-        if let current = self.selection, indexSet.contains(current) {
-            self.selection = min(
-                current,
-                self.servers.count - 1 - indexSet.count
-            )
+        if let server = self.selectedServer,
+           let selectedIndex = self.servers.firstIndex(of: server),
+           indexSet.contains(where: { $0 <= selectedIndex }) {
+            // FIXME: Because this doesn't work, the app will crash after deletion.
```

Comment:
Out of index occurs on deletion.

Reply:
Is this resolved by `EditView` of https://developer.apple.com/tutorials/app-dev-training/passing-data-with-bindings? I'll try it later.